pkg/execution/executor/executor.go

```diff
@@ -806,9 +838,33 @@ func (e *executor) HandlePauses(ctx context.Context, iter state.PauseIterator, e
 				return
 			}
 
+			withEvt := evt.GetEvent()
+			with := withEvt.Map()
+			var runID *ulid.ULID
+
+			isInvokeFunctionOpcode := pause.Opcode != nil && *pause.Opcode == enums.OpcodeInvokeFunction.String()
+			isFnFinishedEvent := withEvt.Name == event.FnFinishedName
+
+			if isInvokeFunctionOpcode && isFnFinishedEvent {
```

Comment:
Can we add a comment or extract this into a function so that it's clearer _why_ we need to do this?

On reading the code, it's not obvious that we need to do this specifically because "function invocation should resume with the function's returned data, not with the event wrapper".